apps/src/templates/sectionProgress/standards/sectionStandardsProgressRedux.js

```diff
@@ -92,6 +100,67 @@ export const lessonsByStandard = state => {
   return lessonsByStandardId;
 };
 
+export function getLessonCompletionStatus(state, stageId) {
+  // A lesson is "completed" by a student if at least 60% of the levels are
+  // completed.
+  const levelsPerLessonCompletionThreshold = 0.6;
+  // A lesson is "complete" for a section if passed by 80% of the students in
+  //the section.
+  const studentsPerSectionCompletionThreshold = 0.8;
+
+  let completionByLesson = {};
+
+  if (
+    state.scriptSelection.scriptId &&
+    state.sectionProgress.scriptDataByScript &&
+    state.sectionProgress.studentLevelProgressByScript &&
+    state.sectionProgress.studentLevelProgressByScript[
+      state.scriptSelection.scriptId
+    ] &&
+    state.teacherSections.sections &&
+    state.teacherSections.selectedSectionId
+  ) {
+    const scriptId = state.scriptSelection.scriptId;
+    const stages = state.sectionProgress.scriptDataByScript[scriptId].stages;
+    const numberStudentsInSection =
+      state.teacherSections.sections[state.teacherSections.selectedSectionId]
+        .studentCount;
+    const levelResultsByStudent =
+      state.sectionProgress.studentLevelProgressByScript[scriptId];
+
+    const studentIds = Object.keys(levelResultsByStudent);
+    stages.forEach(stage => {
```

Comment:
since we are passing in the `stageId` to this method, can we just calculate this for the `stageId` we pass in? Right now, we are populating `completionByLesson` with an entry for every stage, and then only returning the entry for the stage we care about.

Reply:
Great catch, Clare! Updated, thanks.